Assets/Mirror/Authenticators/TimeoutAuthenticator.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Mirror.Authenticators
+{
+
+    /// <summary>
+    /// An authenticator that disconnects connections if they don't
+    /// authenticate within a specified time limit.
+    /// </summary>
+    [AddComponentMenu("Network/Authenticators/TimeoutAuthenticator")]
+    public class TimeoutAuthenticator : NetworkAuthenticator
+    {
+
+        public NetworkAuthenticator authenticator;
+        
+        public int timeout = 60;
+
+        public void Awake()
+        {
+            authenticator.OnClientAuthenticated.AddListener(connection => OnClientAuthenticated.Invoke(connection));
+            authenticator.OnServerAuthenticated.AddListener(connection => OnServerAuthenticated.Invoke(connection));
+        }
+
+
+        public override void OnClientAuthenticate(NetworkConnection conn)
+        {
+            StartCoroutine(BeginClientAuthentication(conn));
```

Comment:
what about Invoke(nameof(CheckTimeout), timeout)? that's less complicated than coroutine

Reply:
2 problems with that:

1) I cannot pass the connection to an Invoke method
2) Can you have multiple invokes in flight?

Reply:
good point, nevermind